service/src/main/kotlin/fi/espoo/evaka/daycare/service/DaycareService.kt

```diff
@@ -56,11 +57,11 @@ class DaycareService {
             groupId = groupId,
             startDate = null,
             endDate = null
-        ).isEmpty()
+        ).isEmpty() && !daycareGroupHasMessages(tx, groupId)
 
         if (!isEmpty) throw Conflict("Cannot delete group which has children placed in it")
```

Comment:
Error message should be updated or even better, have separate messages.
```kt
tx.getPlacements(..).isEmpty() || throw ..
daycareGroupHasMessages(..) || throw ..
```

But. Shouldn't the catch clause already handle the existing fk reference even without an explicit "has messages" query?